### ISSUE TITLE
Speaker volume changed to low when the participant gets a call to his mobile during video chat

### ISSUE THREAD

**Issue by ppavanpasupuleti** [twilio/twilio-video-app-react#182]
**Describe the bug**
Facing an issue when 2 participants are in a video chat , one of the participant uses mobile to have the video chat , gets a call and he rejects the calls and continues to have the video visit. The participant who used mobile for video chat ,suddenly the speaker volume reduced to very low.
To get the volume to normal , the user needs to leave and join the room again or in some cases we need to mute and unmute the volume to get the volume back to normal.

**To Reproduce**
1. Make sure two participants added to the room for the video chat.
2. One participant has to be on mobile(iOS - iPhone).
3. The  participant on the mobile gets a call(someone has to call to mobile) and decline the call. 
4. The speaker volume goes low after the call declines or sometime hit the volume mute button.

**Expected behavior**
The speaker volume should get back to normal if the participant declines the call and joins the video chat back.

**Comment by timmydoza** [comment]
Thanks for the bug report @ppavanpasupuleti!

I'm not able to reproduce this bug on my own (my iPhone is a test device and cannot receive calls). I have some questions though.

How often does this issue occur? Every time? 10% of the time?

When the volume is decreased, does that mean that the user on the iPhone hears quiet audio from the other participant, or does the other participant hear quiet audio from the iPhone participant?  By "speaker volume" do you mean the speaker on the iPhone? If the iPhone user's system volume lowered, can it be increased by pressing the 'volume up' button on the side of the phone?

**Comment by ppavanpasupuleti** [comment]
The issue was not consistent and its kind of less than 5% in iOS .

While testing, I found another issue which is reproducible all the time.

I tried this in iOS phone and Mac PC.

Here are the steps:
     1.  Participant A on mac PC 
     2. Participant B on iOS  Mobile phone on Safari browser with a Bluetooth headset or ear pods.
     3. Both joined room and could hear the audio at both the ends.
     4. Participant B switch off the bluetooth headset or disconnects the ear pods in between the 
          video call.
     5. Participant B could not hear Participant A or some times volume is very low. Participant B 
          tries to increase the mobile volume but audio will not increase. Only after browser refresh 
          the audio volume is back again.
      6. For Participant A , there is no audio issues at all , he can hear Participant B all the time.

**Comment by timmydoza** [comment]
Hey @ppavanpasupuleti - thanks for the info, and apologies for the delay.

I can easily reproduce the issue that you described. It appears to be this bug: https://bugs.webkit.org/show_bug.cgi?id=209594

I tried a workaround mentioned in that ticket and it seems to work. It seems that the audio tracks are paused whenever the headphones are unplugged. If you listen to the `paused` event on the audio element and call `.play()` on the element when it is paused, it seems to fix the issue. For example, here is the `useEffect` function from AudioTrack.tsx:
```
  useEffect(() => {
    audioEl.current = track.attach();
    audioEl.current.setAttribute('data-cy-audio-track-name', track.name);
    audioEl.current.onpause = () => audioEl.current?.play();
    document.body.appendChild(audioEl.current);
    return () => track.detach().forEach(el => el.remove());
  }, [track]);
```

Adding the `audioEl.current.onpause = () => audioEl.current?.play();` line seems to be a good workaround. Can you try this out and let me know if it resolves your issue? Looking forward to hearing your thoughts.

**Comment by ppavanpasupuleti** [comment]
@timmydoza  Thanks for the resolution. It worked.

**Comment by timmydoza** [comment]
Glad to hear! Please let me know if you have any other questions.
